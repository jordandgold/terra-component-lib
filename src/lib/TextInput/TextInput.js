import React, { Component } from "react";
import "./TextInput.scss";
import PropTypes from "prop-types";

class TextInput extends Component {
  render() {
    const { name, label, placeholder, value, inputChange, status } = this.props;
    const statusMessage = {
      error: "has-error",
      success: "has-success"
    };
    return (
      <div
        className={`ter-form-item ${status && statusMessage[status.className]}`}
      >
        {label && (
          <label className="ter-form-item__label" for={name}>
            {label}
          </label>
        )}
        <input
          type="text"
          name={name}
          placeholder={placeholder}
          value={value}
          onChange={e => inputChange(e)}
          className={`ter-input`}
        />
        {status && (
          <p className="ter-form-item__status-message">{status.message}</p>
        )}
      </div>
    );
  }
}

export default TextInput;

TextInput.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  inputChange: PropTypes.func.isRequired,
  status: PropTypes.shape({
    className: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired
  }),
  name: PropTypes.string.isRequired
};
