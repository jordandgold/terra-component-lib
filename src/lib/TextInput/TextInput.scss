@import "../global-styles/terra.scss";

@mixin input-base() {
  font-size: $input-font-size;
  line-height: $input-height;
  height: $input-height;
  color: $input-text-color;
  box-sizing: border-box;
  padding: 0 spacing("xs");
  position: relative;
  z-index: 1;
  display: block;
  width: 100%;
  border-width: $input-border-width;
  border-style: solid;
  border-color: $input-border-color;
  background-color: $input-bg;
  border-radius: $input-border-radius;
  box-sizing: border-box;
  .is-inverse & {
    color: $input-inverse-text-color;
    border-width: $input-border-width;
    border-style: solid;
    border-color: $input-inverse-border-color;
    background-color: $input-inverse-bg;
  }
}

.ter-input {
  @include input-base();
}
