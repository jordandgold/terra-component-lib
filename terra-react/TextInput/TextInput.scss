@import "../global-styles/terra.scss";

.ter-input {
  @include input-base();
}
