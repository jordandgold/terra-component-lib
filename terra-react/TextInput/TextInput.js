import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import React, { Component } from "react";
import "./TextInput.scss";

var TextInput =
/*#__PURE__*/
function (_Component) {
  _inherits(TextInput, _Component);

  function TextInput() {
    _classCallCheck(this, TextInput);

    return _possibleConstructorReturn(this, _getPrototypeOf(TextInput).apply(this, arguments));
  }

  _createClass(TextInput, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          name = _this$props.name,
          label = _this$props.label,
          placeholder = _this$props.placeholder,
          value = _this$props.value,
          inputChange = _this$props.inputChange,
          status = _this$props.status;
      var statusMessage = {
        error: "has-error",
        success: "has-success"
      };
      return React.createElement("div", {
        className: "ter-form-item ".concat(status && statusMessage[status.className])
      }, label && React.createElement("label", {
        className: "ter-form-item__label",
        htmlFor: name
      }, label), React.createElement("input", {
        type: "text",
        name: name,
        placeholder: placeholder,
        value: value,
        onChange: function onChange(e) {
          return inputChange(e);
        },
        className: "ter-input"
      }), status && status.message && React.createElement("p", {
        className: "ter-form-item__status-message"
      }, status.message));
    }
  }]);

  return TextInput;
}(Component);

export default TextInput;