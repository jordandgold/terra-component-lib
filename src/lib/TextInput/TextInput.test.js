import React from "react";
import { shallow } from "enzyme";
import TextInput from "./TextInput";

describe("TextInput", () => {
  let wrapper;
  let mockInputChange;

  beforeEach(() => {
    mockInputChange = jest.fn();
    wrapper = shallow(
      <TextInput
        inputChange={mockInputChange}
        value="mock value"
        name="mock name"
        status=""
        placeholder="mock placeholder"
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call inputChange onChange", () => {
    const mockEvent = {
      target: {
        value: "s"
      }
    };

    wrapper.find(".ter-input").simulate("change", mockEvent);

    expect(mockInputChange).toHaveBeenCalledWith(mockEvent);
  });
});
