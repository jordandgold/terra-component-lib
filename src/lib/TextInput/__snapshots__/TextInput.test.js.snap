// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match the snapshot 1`] = `
<div
  className="ter-form-item undefined"
>
  <label
    className="ter-form-item__label"
    for="mock name"
  >
    mock label
  </label>
  <input
    className="ter-input"
    name="mock name"
    onChange={[Function]}
    placeholder="mock placeholder"
    type="text"
    value="mock value"
  />
</div>
`;

exports[`TextInput should match the snapshot with a status message 1`] = `
<div
  className="ter-form-item has-error"
>
  <label
    className="ter-form-item__label"
    for="mock name"
  >
    mock label
  </label>
  <input
    className="ter-input"
    name="mock name"
    onChange={[Function]}
    placeholder="mock placeholder"
    type="text"
    value="mock value"
  />
  <p
    className="ter-form-item__status-message"
  >
    error!
  </p>
</div>
`;
